File: src/main/java/com/protryon/jasm/util/TrackingByteArrayInputStream.java
package com.protryon.jasm.util;

import java.io.ByteArrayInputStream;

public class TrackingByteArrayInputStream extends ByteArrayInputStream {
    public TrackingByteArrayInputStream(byte[] buf) {
        super(buf);
    }

    public TrackingByteArrayInputStream(byte[] buf, int offset, int length) {
        super(buf, offset, length);
    }

    public int getPosition() {
        return this.pos;
    }
}

File: src/main/java/com/protryon/jasm/util/TrackingByteArrayInputStream.kt
package com.protryon.jasm.util

import java.io.ByteArrayInputStream

class TrackingByteArrayInputStream : ByteArrayInputStream {

    val position: Int
        get() = this.pos

    constructor(buf: ByteArray) : super(buf)

    constructor(buf: ByteArray, offset: Int, length: Int) : super(buf, offset, length)
}
